test/SparqlParserTest.cpp

```diff
@@ -716,13 +717,72 @@ TEST(ParserTest, testParse) {
     }
 
     {
-      // Check if ParseException is thrown after GroupBy with Select *
-      auto pq = SparqlParser(
-          "SELECT DISTINCT * WHERE { \n"
-          "  ?a <b> ?c .\n"
-          "} \n"
-          "GROUP BY ?a ?c \n");
-      ASSERT_THROW(pq.parse(), ParseException);
+      // Check if the correct ParseException is thrown after
+      // GroupBy with Select '*'
+      try {
+        auto pq = SparqlParser(
+                      "SELECT DISTINCT * WHERE { \n"
+                      "  ?a <b> ?c .\n"
+                      "} \n"
+                      "GROUP BY ?a ?c \n")
+                      .parse();
```

Comment:
I think the `ASSERT_THROW()` was sufficient, otherwise the test immediately breaks if we refactor the error message.
If you want to make this check more detailed, the cleanest way would be to implement a `GroupByInAllVariantsSelectedException` which inherits from `ParseException`, but this is not needed.

The current test passes, if no exception is thrown, so at least you should add `FAIL() << "Expected an exception";`
in this line.

(Similar for the test below).